app/assets/src/index.js

```diff
@@ -4,19 +4,24 @@ import 'materialize-css';
 import 'materialize-css/dist/css/materialize.css';
 import 'react-tipsy/dist/react-tipsy.css';
 import 'font-awesome/scss/font-awesome.scss';
+import StringHelper from './helpers/StringHelper';
 import './loader.scss';
+
+// autoload components for react_component method
 const context = require.context('./components', true, /\.(js|jsx)$/);
 const foundComponents = {};
 const contextKeys = context.keys();
 contextKeys.forEach((key) => {
   let a = null;
+  const methodName = StringHelper.baseName(key);
   if (typeof context(key) === 'function') {
     a = context(key);
   } else if (typeof context(key) === 'object') {
     a = context(key).default;
   }
   if (a && a.name) {
-    foundComponents[a.name] = a;
+    // map the correct component name
+    foundComponents[methodName] = a;
```

Comment:
what if methodName is '' or null

Reply:
we are matching only .js or.jsx that exists so it would not be null, but I've added a check for that anyway methodName.